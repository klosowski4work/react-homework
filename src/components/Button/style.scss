@import '../common/style/_colors.scss';
.button {
    cursor: pointer;
    background-color: $color-main;
    padding: 0.5rem 5rem;
    font-family: monospace;
    color: $color-white;
    border: 0;
    border-radius: 3px;
    &--small {
        padding: 0.3rem 2rem;
        font-size: 1.2rem;
    }
    &--grey {
        background-color: $color-dark-grey;
    }
    &--red-reverse {
        background-color: white;
        color: $color-main;
        border: 1px solid $color-main;
        font-weight: bold;
    }
}

.button-link {
    cursor: pointer;
    color: $color-main;
    background-color: transparent;
    color: $color-main;
    border: 0;
    &--grey {
        color: $color-dark-grey;
    }
}