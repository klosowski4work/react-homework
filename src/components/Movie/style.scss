@import '../common/style/_colors.scss';
.movie {
    margin: 2rem;
    color: $color-grey;
    width: 20rem;
    border: 1px solid #eeeeee;
    border-radius: 5px;
    overflow: hidden;
    &__cover,
    &__cover>img {
        width: 20rem;
        height: 30rem;
    }
    &__footer {
        position: relative;
    }
    &__title {
        padding: 0.7rem;
        padding-right: 3.5rem;
        font-size: 1.2rem;
        font-weight: bold;
        height: 4.2rem;
    }
    &__type {
        padding: 0.25rem 0.7rem;
        font-size: 0.8rem;
    }
    &__release-date {
        position: absolute;
        right: 0;
        top: 0;
        border: 1px solid $color-grey;
        margin: 0.5rem;
        padding: 0.25rem 0.65rem;
        line-height: 0.8rem;
        font-size: 0.8rem;
        border-radius: 3px;
    }
    &--empty {
        height: 0px;
        border: 0;
    }
}