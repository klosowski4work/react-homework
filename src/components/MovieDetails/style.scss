@import '../../components/common/style/_colors.scss';
.movie-detail {
    color: white;
    padding: 5rem 0px;
    margin: 0 -5rem;
    display: flex;
    justify-content: space-around;
    &__poster {
        img {
            width: 30rem;
        }
    }
    &__details {
        max-width: 60%;
        padding: 3rem;
    }
    &__header {
        color: $color-main;
        font-size: 4rem;
        display: flex;
    }
    &__score {
        color: seagreen;
        font-size: 1.8rem;
        margin: 0 2rem;
        min-height: 4.2rem;
        min-width: 4.2rem;
        max-height: 4.2rem;
        max-width: 4.2rem;
        border: 1px solid seagreen;
        text-align: center;
        padding-top: 1.2rem;
        border-radius: 50%;
        line-height: 1;
    }
    &__tagline {
        font-size: 2rem;
        color: rgba(255, 255, 255, 0.8);
        line-height: 1.5
    }
    &__date-time {
        font-size: 1.8rem;
        display: flex;
        margin-top: 2rem;
    }
    &__runtime {
        margin-left: 2rem;
    }
    &__description {
        margin-top: 3rem;
        font-size: 1.8rem;
        line-height: 1.2;
        color: rgba(255, 255, 255, 0.8);
    }
}