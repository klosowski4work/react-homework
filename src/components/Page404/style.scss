.page404 {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    justify-content: space-between;
    &__body {
        text-align: center;
        font-size: 5rem;
    }
    &__search-button {
        position: absolute;
        right: 0px;
        top: 0px;
    }
}