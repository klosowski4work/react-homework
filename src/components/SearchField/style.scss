@import '../common/style/_colors.scss';
.search-field {
    &__input {
        width: 100%;
        color: $color-white;
        background: rgba(128, 128, 128, 0.5);
        padding: 1rem;
        font-family: monospace;
        border: 0;
        border-bottom: 1px solid $color-main;
    }
}