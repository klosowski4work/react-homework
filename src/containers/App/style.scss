.app {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    &__body {
        position: relative;
        flex-grow: 1;
    }
}