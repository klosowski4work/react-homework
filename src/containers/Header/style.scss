.header {
    background-size: cover;
    width: 100%;
    min-height: 30rem;
    display: inline-flex;
    justify-content: center;
    background-image: url('/images/net.jpg');
    position: relative;
    &::before {
        content: '';
        background-color: black;
        position: absolute;
        width: 100%;
        height: 100%;
        opacity: 0.5;
    }
    &__container {
        width: 70%;
        min-width: 70rem;
        margin-top: 1.5rem;
        position: relative;
    }
}