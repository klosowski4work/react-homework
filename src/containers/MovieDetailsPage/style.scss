.movie-detail-page {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    &__body {
        position: relative;
        flex-grow: 1;
    }
    &__search-button {
        position: absolute;
        right: 0px;
        top: 0px;
    }
}