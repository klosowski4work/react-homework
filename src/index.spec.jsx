import React from 'react';
import renderer from 'react-test-renderer';
import './index.jsx';

import './style.scss';

jest.disableAutomock();
jest.mock('../images/net.jpg', () => 'image');

describe('Main file', () => {
    it('should render App component in "#app" container', () => {
        const component = renderer.create(<App />).toJSON()
        expect(component).toMatchSnapshot();
    });
});