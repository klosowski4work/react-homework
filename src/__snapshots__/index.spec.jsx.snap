// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main file should render App component in "#app" container 1`] = `
<div
  className="app"
>
  <div
    className="header"
    style={
      Object {
        "backgroundImage": "url('image')",
      }
    }
  >
    <div
      className="header__container"
    >
      <span
        className="logo"
      >
        netflixroulette
      </span>
      <div
        className="search-movie"
      >
        <h2>
          FIND YOUR MOVIE
        </h2>
        <input
          className="search-field__input"
          type="text"
        />
        <div
          className="search-movie__buttons"
        >
          <div
            className="search-movie__buttons-left-side"
          >
            <h3>
              SEARCH BY
            </h3>
            <button
              className="button button--small  "
              onClick={[Function]}
              type="button"
            >
              TITLE
            </button>
            <button
              className="button button--small  button--grey"
              onClick={[Function]}
              type="button"
            >
              GENRE
            </button>
          </div>
          <button
            className="button   "
            onClick={[Function]}
            type="button"
          >
            SEARCH
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="app__body"
  >
    <div
      className="results"
    >
      <div
        className="results__header"
      >
        <span>
          x movies found
        </span>
        <div
          className="sort-by"
        >
          <h3
            className="sort-by__title"
          >
            Sort by
          </h3>
          <button
            className="button-link   "
            onClick={[Function]}
            type="button"
          >
            release date
          </button>
          <button
            className="button-link   "
            onClick={[Function]}
            type="button"
          >
            rating
          </button>
        </div>
      </div>
      <div
        className="results__container"
      >
        <div
          className="movie"
        >
          <div
            className="movie__cover"
          >
            <img
              src=""
            />
          </div>
          <div
            className="movie__footer"
          >
            <div
              className="movie__title"
            >
              Test1
            </div>
            <div
              className="movie__type"
            >
              type
            </div>
            <div
              className="movie__release-date"
            >
              2018
            </div>
          </div>
        </div>
        <div
          className="movie"
        >
          <div
            className="movie__cover"
          >
            <img
              src=""
            />
          </div>
          <div
            className="movie__footer"
          >
            <div
              className="movie__title"
            >
              Test2
            </div>
            <div
              className="movie__type"
            >
              type
            </div>
            <div
              className="movie__release-date"
            >
              2018
            </div>
          </div>
        </div>
        <div
          className="movie"
        >
          <div
            className="movie__cover"
          >
            <img
              src=""
            />
          </div>
          <div
            className="movie__footer"
          >
            <div
              className="movie__title"
            >
              Test3
            </div>
            <div
              className="movie__type"
            >
              type
            </div>
            <div
              className="movie__release-date"
            >
              2018
            </div>
          </div>
        </div>
        <div
          className="movie"
        >
          <div
            className="movie__cover"
          >
            <img
              src=""
            />
          </div>
          <div
            className="movie__footer"
          >
            <div
              className="movie__title"
            >
              Test4
            </div>
            <div
              className="movie__type"
            >
              type
            </div>
            <div
              className="movie__release-date"
            >
              2018
            </div>
          </div>
        </div>
        <div
          className="movie"
        >
          <div
            className="movie__cover"
          >
            <img
              src=""
            />
          </div>
          <div
            className="movie__footer"
          >
            <div
              className="movie__title"
            >
              Test5
            </div>
            <div
              className="movie__type"
            >
              type
            </div>
            <div
              className="movie__release-date"
            >
              2018
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <span
      className="logo"
    >
      netflixroulette
    </span>
  </div>
</div>
`;
